component/src/main/java/com/celements/structEditor/DefaultStructuredDataEditorService.java

```diff
@@ -52,14 +56,12 @@
 
   private Optional<String> getAttributeNameInternal(XWikiDocument cellDoc, XWikiDocument onDoc) {
     List<String> nameParts = new ArrayList<>();
-    Optional<DocumentReference> classRef = getCellClassRef(cellDoc);
-    Optional<String> fieldName = getCellFieldName(cellDoc);
-    if (fieldName.isPresent()) {
-      if (classRef.isPresent()) {
-        nameParts.add(modelUtils.serializeRefLocal(classRef.get()));
+    if (getCellFieldName(cellDoc).isPresent()) {
+      if (getCellClassRef(cellDoc).isPresent()) {
+        nameParts.add(modelUtils.serializeRefLocal(getCellClassRef(cellDoc).get()));
```

Comment:
double use of Method getCellClassRef(cellDoc), would it be better if you store this into a variable like before?